gaurav/src/components/MarkTime.tsx

```diff
@@ -41,39 +52,60 @@ export const MarkTime = () => {
       console.log("error", error);
     }
   };
+
   const handlePunchOut = () => {
     const time = new Date();
     const punchOutTime = time.getTime();
-    const punchInTime = new Date(punchData[punchData.length - 1].time).getTime();
-    console.log("success");
-    if (punchData.length === 0 || punchData[punchData.length - 1].type === "IN") {
-      const newPunchOutData = [
-        ...punchData,
-        {
-          id: punchData.length,
-          time: new Date().toISOString(),
-          type: "OUT",
-        },
-      ];
-      setPunchData(newPunchOutData);
-      localStorage.setItem("punchData", JSON.stringify(newPunchOutData));
-
-      const newNetTime = (punchOutTime - punchInTime) / 1000 + netTime;
-      setNetTime(newNetTime); // time in seconds
-      localStorage.setItem("netTime", JSON.stringify(newNetTime));
+    const punchInTime = new Date(lastPunchData.time).getTime();
+    if ((punchOutTime - punchInTime) / 1000 < 120) {
+      setIsError(true);
+    } else {
+      console.log("success");
+      if (punchData.length === 0 || lastPunchData.type === "IN") {
+        const newPunchOutData = [
+          ...punchData,
+          {
+            id: punchData.length,
+            time: new Date().toISOString(),
+            type: "OUT",
+          },
+        ];
+        setPunchData(newPunchOutData);
+        localStorage.setItem("punchData", JSON.stringify(newPunchOutData));
+        const newNetTime = (punchOutTime - punchInTime) / 1000 + netTime;
+        setNetTime(newNetTime); // time in seconds
+        localStorage.setItem("netTime", JSON.stringify(newNetTime));
+        const newTotalTime = [
+          ...totalTime,
+          {
+            time: newNetTime,
+            date: new Date().toDateString(),
+          },
+        ];
+        setTotalTime(newTotalTime);
+        localStorage.setItem("totalTime", JSON.stringify(newTotalTime));
+        setIsError(false);
+      }
+      setPunchInDisabled(false);
+      setPunchOutDisabled(true);
```

Comment:
Check if setPunchInDisabled and setPunchOutDisabled are complementary. If yes, then try to use single variable.